### PR TITLE
Added logic to validate the existence of jars in liquibase_libs directory prior to concat to classpath

### PR DESCRIPTION
**Pull request by szandany**
## Environment
**Liquibase Version**:
4.7.1
**Liquibase Integration & Version**:
CLI
**Database Vendor & Version**:
Any
**Operating System Type & Version**:
Linux/Unix and Cygwin/GitBash for Windows.
## Pull Request Type
Fix

- [x] Bug fix (non-breaking change which fixes an issue.)
- [ ] Enhancement/New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Added logic to first check/validate if there is a "liquibase_libs" directory in current path AND is there are any jar files in that folder, before attempting to add "path/to/liquibase_libs/jars" to the Classpath.  This fix should work for Linux/Unix systems as well as Linux Emulators like Cygwin/GitBash for Windows.

## Steps To Reproduce
1. Ensure NOT to have the liquibase_libs directory in the relative path or/and have a liquibase_libs directory that doesn't contain any Jar files.
2. Make sure to have liquibase 4.7.1 or above version.
3. Run the liquibase update command with log-level=info on Linux Centos 7 or GitBash for Windows.

## Actual Behavior
Before this PR fix the following error will occur if there is no liquibase_libs directory in the relative path or/and have a liquibase_libs directory that doesn't contain any Jar files.
Log output from `liquibase update --logLevel=info`
`[2022-01-31 11:48:20] INFO [liquibase.lockservice] Successfully acquired change log lock
[2022-01-31 11:48:20] WARNING [liquibase.integration] Cannot create filesystem for url file:/C:/Users/Datical/CLI-Projects/Oracle_CLI/2_hr_test/liquibase_libs/%3f.jar: Illegal char <?> at index 66: C:/Users/Datical/CLI-Projects/Oracle_CLI/2_hr_test/liquibase_libs/?.jar
java.nio.file.InvalidPathException: Illegal char <?> at index 66: C:/Users/Datical/CLI-Projects/Oracle_CLI/2_hr_test/liquibase_libs/?.jar
        at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
        at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
        at sun.nio.fs.WindowsUriSupport.fromUri(WindowsUriSupport.java:165)
        at sun.nio.fs.WindowsFileSystemProvider.getPath(WindowsFileSystemProvider.java:92)
        at java.nio.file.Paths.get(Paths.java:138)
        at liquibase.resource.ClassLoaderResourceAccessor.loadRootPaths(ClassLoaderResourceAccessor.java:67)
        at liquibase.resource.ClassLoaderResourceAccessor.loadRootPaths(ClassLoaderResourceAccessor.java:92)
        at liquibase.resource.ClassLoaderResourceAccessor.init(ClassLoaderResourceAccessor.java:51)
        at liquibase.resource.ClassLoaderResourceAccessor.openStreams(ClassLoaderResourceAccessor.java:108)
`

## Expected/Desired Behavior
Added logic to first check/validate if there is a "liquibase_libs" directory in current path AND is there are any jar files in that folder, before attempting to add "path/to/liquibase_libs/jars" to the Classpath.  
This fix should work for Linux/Unix systems as well as Linux Emulators like Cygwin/GitBash for Windows.

## Fast Track PR Acceptance Checklist:

- [ ] Build is successful and all new and existing tests pass
- [ ] Added [Unit Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)
- [ ] Added [Integration Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)
- [ ] Added [Test Harness Test(s)](https://github.com/liquibase/liquibase-test-harness/pulls)
- [ ] Documentation Updated

## Need Help?
Come chat with us on our [discord channel](https://discord.com/channels/700506481111597066/700506481572839505)

### CODE REVIEW

File: liquibase-dist/src/main/archive/liquibase [hunk 1]
```diff
@@ -33,10 +33,12 @@ if [ -f /usr/bin/cygpath ]; then
   done
 
   # Gather jars from a local liquibase_libs directory
-  for i in liquibase_libs/*.jar; do
-    i=`cygpath --windows "$i"`
-    CP="$CP;$i"
-  done
+  if [[ -f ./liquibase_libs/*.jar ]]
```
Comment (nvoxland) [inline liquibase-dist/src/main/archive/liquibase:36]: This is maybe going to fail if there are multiple jar files in liquibase_libs? (based on https://www.rushworth.us/lisa/?p=3315, I haven't tested it).

That page suggests doing an ls to check the count, but does that seem too hacky? Is there a better way to catch that?

(the same change will have to be applied below)

File: liquibase-dist/src/main/archive/liquibase [hunk 2]
```diff
@@ -54,10 +56,12 @@ else
...
+    CP="$CP""$CP_SEPARATOR""liquibase_libs/"
+    for i in liquibase_libs/*.jar; do
+      CP="$CP""$CP_SEPARATOR""$i"
+    done
+  fi
 
   CP="$CP""$CP_SEPARATOR""$LIQUIBASE_HOME/lib/"
   for i in "$LIQUIBASE_HOME"/lib/*.jar; do
```
Comment (nvoxland) [inline liquibase-dist/src/main/archive/liquibase:67]: We should probably do whatever if block we do for the liquibase_libs into here also? There should usually be a liquibase_home/lib dir, but I maybe remember seeing the same "can't load from classpath .../lib/*.jar" log message too? 

I don't have an example of what we were sometimes seeing off hand to know for sure, though.

**Review by nvoxland — CHANGES_REQUESTED** [review]

**Comment by mcred** [comment]
Would this also be an issue here?
 https://github.com/liquibase/liquibase/blob/f9e341fca13580d91ec24645075a7fdbcc769090/liquibase-dist/src/main/archive/liquibase.bat#L12

**Comment by szandany** [comment]
@nvoxland please see my last commit.  I improved the logic to detect the existence of one or multiple .jar files in the liquibase_libs directory if directory exists.

**Comment by nvoxland** [comment]
As part of #2850 I ended up moving all the logic of finding jars and building up the classpath out of batch/bash and into a new java LiquibaseLauncher wrapper. 

So this should no longer be needed.

**Pull request closed without merging**